subprojects/base-services/src/main/java/org/gradle/util/CollectionUtils.java

```diff
@@ -184,16 +184,16 @@
         return destination;
     }
 
-    public static <R, I> List<R> collect(List<? extends I> list, Transformer<? extends R, ? super I> transformer) {
-        return collect(list, new ArrayList<R>(list.size()), transformer);
+    public static <R, I> Set<R> collect(Set<? extends I> set, Transformer<? extends R, ? super I> transformer) {
+        return collect(set, new HashSet<R>(set.size()), transformer);
```

Comment:
sets can use the provided size, too

Reply:
Why remove the `List` variant? Was it unused?

Reply:
`Iterable` covers it now

Reply:
👍 Can you move back the `Set`-method to its original place to make it more obvious what changed there?

Reply:
Sure, but that's why I've moved it in a separate commit, to make it obvious.